cmd/simulator/worker/worker.go

```diff
@@ -0,0 +1,316 @@
+// (c) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/params"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ava-labs/subnet-evm/cmd/simulator/key"
+	"github.com/ava-labs/subnet-evm/cmd/simulator/metrics"
+)
+
+const (
+	workerKeyDir = ".simulator/keys"
+)
+
+var (
+	transferGasLimit = uint64(21000)
+	transferAmount   = big.NewInt(1)
+	workDelay        = time.Duration(100 * time.Millisecond)
+	retryDelay       = time.Duration(500 * time.Millisecond)
+
+	chainID     *big.Int
+	signer      types.Signer
+	feeCap      *big.Int
+	priorityFee *big.Int
+
+	maxTransferCost  *big.Int
+	requestAmount    *big.Int
+	minFunderBalance *big.Int
+)
+
+func setupVars(cID *big.Int, bFee uint64, pFee uint64) {
+	chainID = cID
+	signer = types.LatestSignerForChainID(chainID)
+	priorityFee = new(big.Int).SetUint64(pFee * params.GWei)
+	feeCap = new(big.Int).Add(new(big.Int).SetUint64(bFee*params.GWei), priorityFee)
+
+	maxTransferCost = new(big.Int).Mul(new(big.Int).SetUint64(transferGasLimit), feeCap)
+	maxTransferCost = new(big.Int).Add(maxTransferCost, transferAmount)
+
+	requestAmount = new(big.Int).Mul(maxTransferCost, big.NewInt(100))
+	minFunderBalance = new(big.Int).Add(maxTransferCost, requestAmount)
+}
+
+func createWorkers(ctx context.Context, keys []*key.Key, endpoints []string, desiredWorkers int) (*worker, []*worker, error) {
+	var master *worker
+	var workers []*worker
+	for _, k := range keys {
+		worker, err := newWorker(k, endpoints[rand.Intn(len(endpoints))])
+		if err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchBalance(ctx); err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchNonce(ctx); err != nil {
+			return nil, nil, fmt.Errorf("could not get nonce: %w", err)
+		}
+		log.Printf("loaded worker %s (balance=%s nonce=%d)\n", worker.k.Address.Hex(), worker.balance.String(), worker.nonce)
+
+		switch {
+		case master == nil:
+			master = worker
+		case new(big.Int).Sub(master.balance, worker.balance).Sign() < 0:
```

Comment:
```suggestion
		master.balance.Cmp(worker.balance) < 0:
```

Reply:
Could we also add a comment explaining that we swap the worker for the master if it has a higher balance so that the master worker ends up with the highest balance?